fix: simplify null and undefined checks in get function

OgabekYuldoshev opened the pull request:
## Summary
This PR fixes the `get` function to properly handle `null` values by returning the default value when the final resolved value is `null`. Previously, the function only checked for `undefined` at the end, allowing `null` values to be returned even when a default value was provided.

**Changes:**
- Updated the final return condition to check for both `undefined` and `null` values
- Ensures consistent behavior where both `null` and `undefined` trigger default value fallback
- Maintains backward compatibility for all existing use cases

This improves the robustness of the utility function and provides more predictable behavior when dealing with nullable data structures.

## Related issue, if any:
<!-- Paste issue's link or number hashtag here if applicable -->

## For any code change,
- [x] Related tests have been added or updated, if needed
- [x] Related documentation has been updated, if needed
- [ ] Related benchmarks have been added or updated, if needed
- [ ] Release notes in [next-minor.md](.github/next-minor.md) or [next-major.md](.github/next-major.md) have been added, if needed

## Does this PR introduce a breaking change?
No

This change maintains backward compatibility. The only behavioral change is that `null` values at the end of a path resolution will now return the default value instead of `null`, which aligns with the expected behavior of most developers using this utility function.

## Bundle impact

| Status | File | Size [^1337] | Difference |
| --- | --- | --- | --- |
| M | `src/object/get.ts` | 167 | -23 (-12%) |

[^1337]: Function size includes the `import` dependencies of the function.

Commit:
fix: simplify null and undefined checks in get function

aleclarson commented on src/object/get.ts:
```diff
@@ -23,10 +23,7 @@ export function get<TDefault = unknown>(
-      return defaultValue as TDefault
-    }
-    if (current === undefined) {
+    if (current === undefined || current === null) {
```
Should use `current == null` to check for both.

OgabekYuldoshev replied:
Thanks for the feedback!
I’ve updated the code to use current == null as suggested so it handles both null and undefined in a concise way.
Added tests for null handling and falsy values to prevent regressions, and updated the docs to mention the new behavior.

aleclarson reviewed (changes requested):
I agree with this change. It makes `get` behave similarly to `a ?? b`.

This is a behavior change, so a test needs to be updated or added such that it will fail when using the previous implementation. This behavior should also be documented.

Commit:
fix(get): treat null as undefined for default value

- Updated get() to return defaultValue when the resolved value is null, matching behavior for undefined.
- Simplified check to use `current == null` for both null and undefined.
- Added unit tests to cover:
  * Returning defaultValue when final value is null
  * Preserving falsy but non-nullish values (0, '', false)
- Updated documentation to note null handling and show examples.

OgabekYuldoshev committed:
Merge branch 'main' into fix-object-get

OgabekYuldoshev committed:
Merge branch 'main' into fix-object-get
